lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt

```diff
@@ -15,44 +15,60 @@ import java.time.format.DateTimeParseException
 
 data class SequenceFilterValue(val type: SequenceFilterFieldType, val value: String, val originalKey: String)
 
+typealias SequenceFilterFieldName = String
+
 @Component
-class SiloFilterExpressionMapper(private val allowedSequenceFilterFields: SequenceFilterFields) {
-    fun map(sequenceFilters: Map<String, String>): SiloFilterExpression {
+class SiloFilterExpressionMapper(
+    private val allowedSequenceFilterFields: SequenceFilterFields,
+    private val variantQueryFacade: VariantQueryFacade,
+) {
+    fun map(sequenceFilters: Map<SequenceFilterFieldName, String>): SiloFilterExpression {
         if (sequenceFilters.isEmpty()) {
             return True
         }
 
-        val filterExpressions = sequenceFilters
+        val allowedSequenceFiltersWithType = sequenceFilters
             .map { (key, value) ->
                 val nullableType = allowedSequenceFilterFields.fields[key]
                 val (filterExpressionId, type) = mapToFilterExpressionIdentifier(nullableType, key)
                 filterExpressionId to SequenceFilterValue(type, value, key)
             }
             .groupBy({ it.first }, { it.second })
-            .map { (key, values) ->
-                val (siloColumnName, siloFilterPrimitive) = key
-                when (siloFilterPrimitive) {
-                    SiloFilterPrimitive.StringEquals -> StringEquals(siloColumnName, values[0].value)
-                    SiloFilterPrimitive.PangoLineage -> mapToPangoLineageFilter(siloColumnName, values[0].value)
-                    SiloFilterPrimitive.DateBetween -> mapToDateBetweenFilter(siloColumnName, values)
-                    SiloFilterPrimitive.NucleotideSymbolEquals -> mapToNucleotideFilter(values[0].value)
-                }
+
+        if (allowedSequenceFiltersWithType.keys.any { it.second == Filter.VariantQuery } &&
+            allowedSequenceFiltersWithType.keys.any { it.second in variantQueryTypes }
+        ) {
+            throw IllegalArgumentException(
+                "variantQuery cannot be used with other variant filters",
```

Comment:
Can we find an exception message that explains a bit better why the exception is thrown? I can imagine that an unsuspecting user won't get what they did wrong.